kasa/discover.py

```diff
@@ -331,20 +340,25 @@ async def connect_single(
         The device type is discovered by querying the device.
 
         :param host: Hostname of device to query
+        :param device_type: Device type to use for the device.
         :rtype: SmartDevice
         :return: Object for querying/controlling found device.
         """
-        unknown_dev = SmartDevice(
-            host=host, port=port, credentials=credentials, timeout=timeout
-        )
-        await unknown_dev.update()
-        device_class = Discover._get_device_class(unknown_dev.internal_state)
-        dev = device_class(
-            host=host, port=port, credentials=credentials, timeout=timeout
-        )
-        # Reuse the connection from the unknown device
-        # so we don't have to reconnect
-        dev.protocol = unknown_dev.protocol
+        if device_type and (klass := DEVICE_TYPE_TO_CLASS.get(device_type)):
+            dev = klass(host=host, port=port, credentials=credentials, timeout=timeout)
```

Comment:
```suggestion
            dev = klass(host=host, port=port, credentials=credentials, timeout=timeout)
            await dev.update()
            return dev
```
Let's return early here to avoid the else block and nesting. If you want, add a quick note that if the device type is given the there's no  auto-detection.

Reply:
adjusted in bcdcd5f982bfccf39c33c03300ebdaeb88d8dd09